runtime/browser/ui/desktop/exclusive_access_bubble.h

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) 2012 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef XWALK_RUNTIME_BROWSER_UI_DESKTOP_EXCLUSIVE_ACCESS_BUBBLE_H_
+#define XWALK_RUNTIME_BROWSER_UI_DESKTOP_EXCLUSIVE_ACCESS_BUBBLE_H_
+
+#include "base/timer/timer.h"
+#include "ui/gfx/animation/animation_delegate.h"
+#include "ui/gfx/geometry/point.h"
+#include "url/gurl.h"
+
+namespace gfx {
+class Rect;
+}
+
+// Describes the contents of the fullscreen exit bubble.
+enum ExclusiveAccessBubbleType {
+  EXCLUSIVE_ACCESS_BUBBLE_TYPE_NONE = 0,
+
+  // For browser fullscreen mode.
+  EXCLUSIVE_ACCESS_BUBBLE_TYPE_BROWSER_FULLSCREEN_EXIT_INSTRUCTION,
+  // For closing the application.
+  EXCLUSIVE_ACCESS_BUBBLE_TYPE_APPLICATION_FULLSCREEN_CLOSE_INSTRUCTION
+};
+
+// Bubble that informs the user when an exclusive access state is in effect and
+// as to how to exit out of the state. Currently there are two exclusive access
+// state, namely fullscreen and mouse lock.
+//
+// Notification display design note: if the #simplified-fullscreen-ui flag is
+// enabled, the bubble has the following behaviour:
+// - The bubble is shown for kInitialDelayMs, then hides.
+// - After a bubble has been shown, notifications are suppressed for
+//   kSnoozeNotificationsTimeMs, to avoid bothering the user. After this time
+//   has elapsed, the next user input re-displays the bubble.
+class ExclusiveAccessBubble : public gfx::AnimationDelegate {
+ public:
+  explicit ExclusiveAccessBubble(const GURL& url,
```

Comment:
`explicit ` is meaningless since we have more than 1 arg here.
Do we actually need `url` arg? how is it used?